### PR TITLE
Controller thread dies somehow, prevent that

### PR DESCRIPTION
**Pull request by bserdar**

### CODE REVIEW

File: migrator/src/main/java/com/redhat/lightblue/migrator/Controller.java [hunk 1]
```diff
@@ -28,7 +28,7 @@
     private final LightblueClient lightblueClient;
     private final Map<String, MigrationProcess> migrationMap = new HashMap<>();
     private final ThreadMonitor threadMonitor;
-    private boolean stopped=false;
+    private volatile boolean stopped=false;
```
Comment (dcrissman) [inline migrator/src/main/java/com/redhat/lightblue/migrator/Controller.java]: I know what volatile does, but what is the reason for using it here?

File: migrator/src/main/java/com/redhat/lightblue/migrator/Controller.java [hunk 2]
```diff
@@ -224,15 +224,10 @@ public void run() {
                 MigrationConfiguration[] cfg=getMigrationConfigurations();
                 createControllers(cfg);
                 Breakpoint.checkpoint("Controller:createconfig");
-            } catch (Exception e) {
+                Thread.sleep(30000);
+            } catch (Throwable e) {
```
Comment (dcrissman) [inline migrator/src/main/java/com/redhat/lightblue/migrator/Controller.java]: Why catch the InterruptedException here? There is no need to log it as it simply means the thread has been terminated.

File: migrator/src/main/java/com/redhat/lightblue/migrator/Controller.java (hunk 1 above)
Comment (bserdar) [inline migrator/src/main/java/com/redhat/lightblue/migrator/Controller.java]: Agreed, no need.

File: migrator/src/main/java/com/redhat/lightblue/migrator/Controller.java (hunk 2 above)
Comment (bserdar) [inline migrator/src/main/java/com/redhat/lightblue/migrator/Controller.java]: This thread somehow terminates. Any exception thrown here, including InterruptedException, doesn't mean the thread should terminate. It should only terminate when stopped is set.

**Pull request merged**